File: Iliya/kafka js/topic.js
const {Kafka} = require("kafkajs")



run();



async function run(){
    try
    {
        const kafka = new Kafka({
            "clientId": "yeet",
            "brokers" :["localhost:9093", "localhost:9094", "localhost:9095"]
       })

       const admin = kafka.admin();
       console.log("Connecting.....")
       await admin.connect()
       console.log("Connected!")


       await admin.createTopics({

            "topics": [{

                "topic": "yeet",
                "numPartitions": 2,
                "replicationFactor": 2
            }, {
                "topic": "something",
                "numPartitions": 2,
                "replicationFactor": 1
            }]
        })
        console.log("Nothing wrong so far")
        await admin.disconnect();




    }
    catch(ex)
    {
        console.error(`Error code ${ex}`)
    }

    finally{
        process.exit(0);
    }
}

File: Iliya/kafka python but CLOUD/topic.py
#Topic.py

from kafka import KafkaAdminClient

from kafka.admin import NewTopic



try:
    client_adm = KafkaAdminClient(bootstrap_servers=['localhost:9093', 'localhost:9094', 'localhost:9095'])
    
    yeet = NewTopic(
        name='yeet', 
        num_partitions=2, 
        replication_factor=2
    )
    
    something = NewTopic(
        name='something', 
        num_partitions=2,
        replication_factor=1
    )
    
    last = NewTopic(
        name='last', 
        num_partitions=2
        ,
        replication_factor=3
    )

    client_adm.create_topics(new_topics=[yeet, something, last])
    print ("We got this far")

    
except Exception as error:
        
        
    print("Error messsage:  ", error)

